### PR TITLE
fix(nestjs): Lookup request from GraphQL context in ArcjetGuard

### LINKED ISSUE #1856: NestJS: Support looking up request from graphql context
NestJS supports GraphQL with an official package. When GraphQL is used, the request must be accessed in a different fashion, see https://github.com/nestjs/graphql/blob/8d19548dd8cb8c6d6003552673a6646603d2e22f/packages/graphql/lib/services/gql-execution-context.ts

Our guard should be able to look up the request in both the http and graphql contexts.

### PR DESCRIPTION
**Pull request by blaine-arcjet**
This adds support for using ArcjetGuard globally on GraphQL applications. With this, we'll check the context type and lookup the request if it is `"http"` or `"graphql"`.

I've also added an example for using ArcjetGuard with a NestJS GraphQL server.

Closes #1856

### CODE REVIEW

File: arcjet-nest/index.ts [hunk 1]
```diff
@@ -395,7 +433,12 @@ let ArcjetGuard = class ArcjetGuard implements CanActivate {
       aj = rules.reduce((aj, rule) => aj.withRule(rule), aj);
     }
 
-    const request = context.switchToHttp().getRequest();
+    const request = requestFromContext(context);
+
+    // If we cannot access the request, we "fail open" by allowing the request
+    if (typeof request === "undefined") {
+      return true;
```
Comment (davidmytton) [inline arcjet-nest/index.ts:440]: Should we log a warning in this fail open case?

Comment (blaine-arcjet) [reply arcjet-nest/index.ts:440]: We don't have access to the logger in the guard.

**Comment by blaine-arcjet** [comment]
@SocketSecurity ignore npm/@protobufjs/fetch@1.1.0
@SocketSecurity ignore npm/async-retry@1.3.3
@SocketSecurity ignore npm/@apollo/protobufjs@1.2.7
@SocketSecurity ignore npm/subscriptions-transport-ws@0.11.0
@SocketSecurity ignore npm/@apollo/server-plugin-landing-page-graphql-playground@4.0.0
@SocketSecurity ignore npm/ws@8.17.1
@SocketSecurity ignore npm/ws@7.5.10
@SocketSecurity ignore npm/@apollo/server@4.11.0

File: arcjet-nest/index.ts (hunk 1 above)
Comment (blaine-arcjet) [reply arcjet-nest/index.ts:440]: https://github.com/arcjet/arcjet-js/issues/1859

**Review by davidmytton — APPROVED** [review]

**Pull request merged**